src/coreclr/vm/tieredcompilation.cpp

```diff
@@ -1018,7 +1043,20 @@ CORJIT_FLAGS TieredCompilationManager::GetJitFlags(PrepareCodeConfig *config)
         {
             if (g_pConfig->TieredCompilation_QuickJit())
             {
-                _ASSERTE(nativeCodeVersion.GetOptimizationTier() == NativeCodeVersion::OptimizationTier0);
+                if (nativeCodeVersion.GetOptimizationTier() == NativeCodeVersion::OptimizationTier::OptimizationTier0Instrumented)
+                {
+                    flags.Set(CORJIT_FLAGS::CORJIT_FLAG_BBINSTR);
+                    if (CLRConfig::GetConfigValue(CLRConfig::UNSUPPORTED_TC_OptimizedInstrumentedTier) != 0)
+                    {
+                        flags.Set(CORJIT_FLAGS::CORJIT_FLAG_TIER1);
```

Comment:
I think this decision should be driven by a different native code version. We'd like to be able to determine in SOS/ETW what kind of code was produced (in particular, if the method was optimized).

Reply:
Done